[Automation] javac compile fails for org.jboss.arquillian.container:arquillian-container-test-api:1.1.11.Final

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25302393571/job/74171619172
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final
Raw versions JSON: [
  "1.1.11.Final",
  "1.1.12.Final",
  "1.1.13.Final",
  "1.1.14.Final",
  "1.1.15.Final",
  "1.2.0.Final",
  "1.2.1.Final",
  "1.3.0.Final",
  "1.4.0.Final",
  "1.4.1.Final",
  "1.5.0.Final",
  "1.6.0.Final",
  "1.7.0.Final",
  "1.7.1.Final",
  "1.7.2.Final",
  "1.8.0.Final",
  "1.8.1.Final",
  "1.9.0.Final",
  "1.9.1.Final",
  "1.9.2.Final",
  "1.9.3.Final",
  "1.9.4.Final",
  "1.9.5.Final",
  "1.10.0.Final",
  "1.10.1.Final"
]
Parsed version count: 25
Parsed versions: 1.1.11.Final 1.1.12.Final 1.1.13.Final 1.1.14.Final 1.1.15.Final 1.2.0.Final 1.2.1.Final 1.3.0.Final 1.4.0.Final 1.4.1.Final 1.5.0.Final 1.6.0.Final 1.7.0.Final 1.7.1.Final 1.7.2.Final 1.8.0.Final 1.8.1.Final 1.9.0.Final 1.9.1.Final 1.9.2.Final 1.9.3.Final 1.9.4.Final 1.9.5.Final 1.10.0.Final 1.10.1.Final
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final
========================================================================================
========================================================================================
 org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final stage native-image run
========================================================================================
Starting stage 'native-image run' for 1.1.11.Final at 2026-05-04T05:21:18Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean test -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
    private static final class RecordingArchive implements Archive<RecordingArchive> {
                         ^
1 error


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
      private static final class RecordingArchive implements Archive<RecordingArchive> {
                           ^
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
BUILD FAILED in 5s
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 14s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.1.11.Final at 2026-05-04T05:21:33Z with exit code 1 after 15s
❌ ERROR: 'native-image run' for 1.1.11.Final failed with exit code 1.
SOFT-FAIL[native-image run][1.1.11.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean test -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"]
========================================================================================
 org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.1.11.Final at 2026-05-04T05:21:33Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 901ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
    private static final class RecordingArchive implements Archive<RecordingArchive> {
                         ^
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
      private static final class RecordingArchive implements Archive<RecordingArchive> {
                           ^
  1 error

* Try:

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.


BUILD FAILED in 1s
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.1.11.Final at 2026-05-04T05:21:36Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 1.1.11.Final failed with exit code 1.
FAILED[javac compile][1.1.11.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25302393571/job/74171619179
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final
Raw versions JSON: [
  "1.1.11.Final",
  "1.1.12.Final",
  "1.1.13.Final",
  "1.1.14.Final",
  "1.1.15.Final",
  "1.2.0.Final",
  "1.2.1.Final",
  "1.3.0.Final",
  "1.4.0.Final",
  "1.4.1.Final",
  "1.5.0.Final",
  "1.6.0.Final",
  "1.7.0.Final",
  "1.7.1.Final",
  "1.7.2.Final",
  "1.8.0.Final",
  "1.8.1.Final",
  "1.9.0.Final",
  "1.9.1.Final",
  "1.9.2.Final",
  "1.9.3.Final",
  "1.9.4.Final",
  "1.9.5.Final",
  "1.10.0.Final",
  "1.10.1.Final"
]
Parsed version count: 25
Parsed versions: 1.1.11.Final 1.1.12.Final 1.1.13.Final 1.1.14.Final 1.1.15.Final 1.2.0.Final 1.2.1.Final 1.3.0.Final 1.4.0.Final 1.4.1.Final 1.5.0.Final 1.6.0.Final 1.7.0.Final 1.7.1.Final 1.7.2.Final 1.8.0.Final 1.8.1.Final 1.9.0.Final 1.9.1.Final 1.9.2.Final 1.9.3.Final 1.9.4.Final 1.9.5.Final 1.10.0.Final 1.10.1.Final
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final
========================================================================================
========================================================================================
 org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final stage native-image run
========================================================================================
Starting stage 'native-image run' for 1.1.11.Final at 2026-05-04T05:21:27Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean test -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 5s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
    private static final class RecordingArchive implements Archive<RecordingArchive> {
                         ^
1 error

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
      private static final class RecordingArchive implements Archive<RecordingArchive> {
                           ^
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 3s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 10s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.1.11.Final at 2026-05-04T05:21:37Z with exit code 1 after 10s
❌ ERROR: 'native-image run' for 1.1.11.Final failed with exit code 1.
SOFT-FAIL[native-image run][1.1.11.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean test -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"]
========================================================================================
 org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.1.11.Final at 2026-05-04T05:21:37Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 702ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
    private static final class RecordingArchive implements Archive<RecordingArchive> {
                         ^
1 error
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
FAILURE: Build failed with an exception.
10 actionable tasks: 1 executed, 9 up-to-date

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
      private static final class RecordingArchive implements Archive<RecordingArchive> {
                           ^
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 953ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.1.11.Final at 2026-05-04T05:21:40Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 1.1.11.Final failed with exit code 1.
FAILED[javac compile][1.1.11.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25302393571/job/74171619159
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final
Raw versions JSON: [
  "1.1.11.Final",
  "1.1.12.Final",
  "1.1.13.Final",
  "1.1.14.Final",
  "1.1.15.Final",
  "1.2.0.Final",
  "1.2.1.Final",
  "1.3.0.Final",
  "1.4.0.Final",
  "1.4.1.Final",
  "1.5.0.Final",
  "1.6.0.Final",
  "1.7.0.Final",
  "1.7.1.Final",
  "1.7.2.Final",
  "1.8.0.Final",
  "1.8.1.Final",
  "1.9.0.Final",
  "1.9.1.Final",
  "1.9.2.Final",
  "1.9.3.Final",
  "1.9.4.Final",
  "1.9.5.Final",
  "1.10.0.Final",
  "1.10.1.Final"
]
Parsed version count: 25
Parsed versions: 1.1.11.Final 1.1.12.Final 1.1.13.Final 1.1.14.Final 1.1.15.Final 1.2.0.Final 1.2.1.Final 1.3.0.Final 1.4.0.Final 1.4.1.Final 1.5.0.Final 1.6.0.Final 1.7.0.Final 1.7.1.Final 1.7.2.Final 1.8.0.Final 1.8.1.Final 1.9.0.Final 1.9.1.Final 1.9.2.Final 1.9.3.Final 1.9.4.Final 1.9.5.Final 1.10.0.Final 1.10.1.Final
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final
========================================================================================
========================================================================================
 org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final stage native-image run
========================================================================================
Starting stage 'native-image run' for 1.1.11.Final at 2026-05-04T05:21:07Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.1.11.Final" ./gradlew clean test -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
    private static final class RecordingArchive implements Archive<RecordingArchive> {
                         ^
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html

1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
      private static final class RecordingArchive implements Archive<RecordingArchive> {
                           ^
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 4s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date


> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 12s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.1.11.Final at 2026-05-04T05:21:20Z with exit code 1 after 13s
❌ ERROR: 'native-image run' for 1.1.11.Final failed with exit code 1.
SOFT-FAIL[native-image run][1.1.11.Final][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.1.11.Final" ./gradlew clean test -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"]
========================================================================================
 org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final:1.1.11.Final stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.1.11.Final at 2026-05-04T05:21:20Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 886ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
    private static final class RecordingArchive implements Archive<RecordingArchive> {
                         ^
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.10.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java:300: error: RecordingArchive is not abstract and does not override abstract method shallowCopy(Filter<ArchivePath>) in Archive
      private static final class RecordingArchive implements Archive<RecordingArchive> {
                           ^
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.1.11.Final at 2026-05-04T05:21:23Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 1.1.11.Final failed with exit code 1.
FAILED[javac compile][1.1.11.Final][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.1.11.Final" ./gradlew clean compileTestJava -Pcoordinates="org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final"]
```

--- a/metadata/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/reachability-metadata.json
+++ b/metadata/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/reachability-metadata.json
@@ -1,0 +1,107 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.jboss.arquillian.container.test.api.Testable"
+      },
+      "type" : "org.jboss.shrinkwrap.impl.base.path.BasicPath",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "org.jboss.shrinkwrap.api.ArchivePath",
+            "org.jboss.shrinkwrap.api.ArchivePath"
+          ]
+        },
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "org.jboss.shrinkwrap.api.ArchivePath",
+            "java.lang.String"
+          ]
+        },
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.lang.String",
+            "org.jboss.shrinkwrap.api.ArchivePath"
+          ]
+        },
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.lang.String",
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.jboss.shrinkwrap.api.ShrinkWrap"
+      },
+      "type" : "org.jboss.shrinkwrap.impl.base.ServiceExtensionLoader",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.lang.Iterable"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.jboss.shrinkwrap.api.ShrinkWrap"
+      },
+      "type" : "org.jboss.shrinkwrap.impl.base.MemoryMapArchiveImpl",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "org.jboss.shrinkwrap.api.Configuration"
+          ]
+        },
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.lang.String",
+            "org.jboss.shrinkwrap.api.Configuration"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.jboss.shrinkwrap.api.ShrinkWrap"
+      },
+      "type" : "org.jboss.shrinkwrap.impl.base.spec.JavaArchiveImpl",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "org.jboss.shrinkwrap.api.Archive"
+          ]
+        }
+      ]
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.jboss.shrinkwrap.api.ShrinkWrap"
+      },
+      "glob" : "META-INF/services/org.jboss.shrinkwrap.api.spec.JavaArchive"
+    }
+  ]
+}

--- a/metadata/org.jboss.arquillian.container/arquillian-container-test-api/index.json
+++ b/metadata/org.jboss.arquillian.container/arquillian-container-test-api/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "1.1.11.Final",
+    "source-code-url" : "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-sources.jar",
+    "repository-url" : "https://github.com/arquillian/arquillian-core",
+    "test-code-url" : "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-tests.jar",
+    "documentation-url" : "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-javadoc.jar",
+    "description" : "Arquillian Container Test API provides annotations and helper APIs used to declare deployments, target containers, run modes, and protocols for Arquillian-managed integration tests. It lets tests coordinate container startup, deployment, and client-vs-container execution through contracts such as Deployment, RunAsClient, TargetsContainer, Deployer, and ContainerController.",
     "tested-versions" : [
       "1.1.11.Final"
     ],

--- a/metadata/org.jboss.arquillian.container/arquillian-container-test-api/index.json
+++ b/metadata/org.jboss.arquillian.container/arquillian-container-test-api/index.json
@@ -1,16 +1,26 @@
 [
   {
-    "latest": true,
-    "metadata-version": "1.1.10.Final",
-    "source-code-url": "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-sources.jar",
-    "repository-url": "https://github.com/arquillian/arquillian-core",
-    "test-code-url": "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-tests.jar",
-    "documentation-url": "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-javadoc.jar",
-    "description": "Arquillian Container Test API provides annotations and helper APIs used to declare deployments, target containers, run modes, and protocols for Arquillian-managed integration tests. It lets tests coordinate container startup, deployment, and client-vs-container execution through contracts such as Deployment, RunAsClient, TargetsContainer, Deployer, and ContainerController.",
-    "tested-versions": [
+    "latest" : true,
+    "metadata-version" : "1.1.11.Final",
+    "tested-versions" : [
+      "1.1.11.Final"
+    ],
+    "allowed-packages" : [
+      "org.jboss.arquillian.container.test.api",
+      "org.jboss.shrinkwrap.api"
+    ]
+  },
+  {
+    "metadata-version" : "1.1.10.Final",
+    "source-code-url" : "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-sources.jar",
+    "repository-url" : "https://github.com/arquillian/arquillian-core",
+    "test-code-url" : "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-tests.jar",
+    "documentation-url" : "https://repo1.maven.org/maven2/org/jboss/arquillian/container/arquillian-container-test-api/$version$/arquillian-container-test-api-$version$-javadoc.jar",
+    "description" : "Arquillian Container Test API provides annotations and helper APIs used to declare deployments, target containers, run modes, and protocols for Arquillian-managed integration tests. It lets tests coordinate container startup, deployment, and client-vs-container execution through contracts such as Deployment, RunAsClient, TargetsContainer, Deployer, and ContainerController.",
+    "tested-versions" : [
       "1.1.10.Final"
     ],
-    "allowed-packages": [
+    "allowed-packages" : [
       "org.jboss.arquillian.container.test.api",
       "org.jboss.shrinkwrap.api"
     ]

--- a/stats/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/execution-metrics.json
+++ b/stats/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/execution-metrics.json
@@ -1,0 +1,87 @@
+{
+  "fix_javac_fail:2026-05-04": {
+    "timestamp": "2026-05-04T07:09:02.776591Z",
+    "library": "org.jboss.arquillian.container:arquillian-container-test-api:1.1.11.Final",
+    "starting_commit": "7f350be7716c2480627bec9d81385a3398db9181",
+    "ending_commit": "05bcb75b98c03c24e1bdec0e90d088fed4a40301",
+    "previous_library": "org.jboss.arquillian.container:arquillian-container-test-api:1.1.10.Final",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "1.1.11.Final",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 35,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 35
+        },
+        "line": {
+          "covered": 10,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 10
+        },
+        "method": {
+          "covered": 7,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 7
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "1.1.10.Final",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 35,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 35
+        },
+        "line": {
+          "covered": 10,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 10
+        },
+        "method": {
+          "covered": 7,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 7
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 24351,
+      "cached_input_tokens_used": 135168,
+      "output_tokens_used": 1642,
+      "iterations": 2,
+      "cost_usd": 0.1169,
+      "tested_library_loc": 10,
+      "metadata_entries": 15,
+      "previous_library_metadata_entries": 15,
+      "code_coverage_percent": 100.0,
+      "previous_library_coverage_percent": 100.0
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java",
+      "metadata_file": "metadata/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/stats.json
+++ b/stats/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "1.1.11.Final",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 35,
+        "missed" : 0,
+        "ratio" : 1.0,
+        "total" : 35
+      },
+      "line" : {
+        "covered" : 10,
+        "missed" : 0,
+        "ratio" : 1.0,
+        "total" : 10
+      },
+      "method" : {
+        "covered" : 7,
+        "missed" : 0,
+        "ratio" : 1.0,
+        "total" : 7
+      }
+    }
+  } ]
+}

--- a/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/.gitignore
+++ b/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/build.gradle
+++ b/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/build.gradle
@@ -1,0 +1,28 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.jboss.arquillian.container:arquillian-container-test-api:$libraryVersion"
+    testImplementation 'org.jboss.shrinkwrap:shrinkwrap-impl-base:1.2.2'
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/gradle.properties
+++ b/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.jboss.arquillian.container:arquillian-container-test-api:1.1.11.Final
+library.version = 1.1.11.Final
+metadata.dir = org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/

--- a/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/settings.gradle
+++ b/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.jboss.arquillian.container.arquillian-container-test-api_tests'

--- a/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java
+++ b/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java
@@ -1,0 +1,577 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_jboss_arquillian_container.arquillian_container_test_api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.arquillian.container.test.api.Config;
+import org.jboss.arquillian.container.test.api.ContainerController;
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.OverProtocol;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.container.test.api.ShouldThrowException;
+import org.jboss.arquillian.container.test.api.TargetsContainer;
+import org.jboss.arquillian.container.test.api.Testable;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.Assignable;
+import org.jboss.shrinkwrap.api.ArchiveEventHandler;
+import org.jboss.shrinkwrap.api.ArchiveFormat;
+import org.jboss.shrinkwrap.api.ArchivePath;
+import org.jboss.shrinkwrap.api.Filter;
+import org.jboss.shrinkwrap.api.Node;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.Asset;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.NamedAsset;
+import org.jboss.shrinkwrap.api.exporter.StreamExporter;
+import org.jboss.shrinkwrap.api.formatter.Formatter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+
+public class Arquillian_container_test_apiTest {
+    @Test
+    void configProvidesMutableFluentPropertyMap() {
+        Config config = new Config();
+
+        Config returned = config.add("bind.address", "127.0.0.1")
+                .add("http.port", "8080")
+                .add("http.port", "8181");
+
+        assertThat(returned).isSameAs(config);
+        assertThat(config.map()).isSameAs(config.getProperties());
+        assertThat(config.map())
+                .containsEntry("bind.address", "127.0.0.1")
+                .containsEntry("http.port", "8181")
+                .hasSize(2);
+
+        config.getProperties().put("debug", "true");
+        assertThat(config.map()).containsEntry("debug", "true");
+    }
+
+    @Test
+    void containerControllerContractAcceptsPlainAndConfiguredLifecycleOperations() {
+        RecordingContainerController controller = new RecordingContainerController();
+        Config config = new Config().add("host", "localhost").add("port", "9090");
+
+        controller.start("primary");
+        controller.start("secondary", config.map());
+        controller.stop("primary");
+        controller.kill("secondary");
+
+        assertThat(controller.isStarted("primary")).isFalse();
+        assertThat(controller.isStarted("secondary")).isFalse();
+        assertThat(controller.events).containsExactly(
+                "start:primary",
+                "start:secondary",
+                "stop:primary",
+                "kill:secondary");
+        assertThat(controller.configurations.get("primary")).isEmpty();
+        assertThat(controller.configurations.get("secondary"))
+                .containsEntry("host", "localhost")
+                .containsEntry("port", "9090");
+    }
+
+    @Test
+    void deployerContractExposesManualDeploymentLifecycleAndPayload() throws Exception {
+        RecordingDeployer deployer = new RecordingDeployer();
+        deployer.register("orders", "archive-bytes");
+
+        deployer.deploy("orders");
+        String payload;
+        try (InputStream stream = deployer.getDeployment("orders")) {
+            payload = new String(stream.readAllBytes(), StandardCharsets.UTF_8);
+        }
+        deployer.undeploy("orders");
+
+        assertThat(payload).isEqualTo("archive-bytes");
+        assertThat(deployer.deployed).doesNotContain("orders");
+        assertThat(deployer.events).containsExactly("deploy:orders", "get:orders", "undeploy:orders");
+    }
+
+    @Test
+    void testableMarksArchiveAndKeepsArchiveType() {
+        RecordingArchive archive = new RecordingArchive("web.war");
+
+        assertThat(Testable.isArchiveToTest(archive)).isFalse();
+        RecordingArchive returned = Testable.archiveToTest(archive);
+
+        assertThat(returned).isSameAs(archive);
+        assertThat(Testable.isArchiveToTest(archive)).isTrue();
+        assertThat(archive.addedPaths).containsExactly(Testable.MARKER_FILE_PATH.get());
+        assertThat(archive.addedAssets).hasSize(1);
+        assertThat(Testable.MARKER_FILE_PATH.get()).isEqualTo("/META-INF/arquillian.ArchiveUnderTest");
+    }
+
+    @Test
+    void testableAddsRetrievableEmptyMarkerAssetToRealArchive() throws Exception {
+        JavaArchive archive = ShrinkWrap.create(JavaArchive.class, "sample.jar");
+
+        JavaArchive returned = Testable.archiveToTest(archive);
+        Node marker = archive.get(Testable.MARKER_FILE_PATH);
+
+        assertThat(returned).isSameAs(archive);
+        assertThat(Testable.isArchiveToTest(archive)).isTrue();
+        assertThat(marker).isNotNull();
+        assertThat(marker.getPath()).isEqualTo(Testable.MARKER_FILE_PATH);
+
+        byte[] markerBytes;
+        try (InputStream markerStream = marker.getAsset().openStream()) {
+            markerBytes = markerStream.readAllBytes();
+        }
+        assertThat(markerBytes).isEmpty();
+    }
+
+    @Test
+    void testableRecognizesArchiveMarkedWithPublicMarkerPath() {
+        JavaArchive archive = ShrinkWrap.create(JavaArchive.class, "pre-marked.jar");
+
+        archive.add(EmptyAsset.INSTANCE, Testable.MARKER_FILE_PATH);
+
+        assertThat(Testable.isArchiveToTest(archive)).isTrue();
+    }
+
+    @Test
+    void deploymentAnnotationsExposeDefaultAndCustomizedValuesAtRuntime() throws Exception {
+        Method defaultDeployment = DeploymentFixture.class.getDeclaredMethod("defaultDeployment");
+        Method manualDeployment = DeploymentFixture.class.getDeclaredMethod("manualDeployment");
+
+        Deployment defaultAnnotation = defaultDeployment.getAnnotationsByType(Deployment.class)[0];
+        Deployment manualAnnotation = manualDeployment.getAnnotationsByType(Deployment.class)[0];
+
+        assertThat(defaultAnnotation.name()).isEqualTo("_DEFAULT_");
+        assertThat(defaultAnnotation.managed()).isTrue();
+        assertThat(defaultAnnotation.order()).isEqualTo(-1);
+        assertThat(defaultAnnotation.testable()).isTrue();
+
+        assertThat(manualAnnotation.name()).isEqualTo("orders");
+        assertThat(manualAnnotation.managed()).isFalse();
+        assertThat(manualAnnotation.order()).isEqualTo(7);
+        assertThat(manualAnnotation.testable()).isFalse();
+        assertRuntimeMethodTarget(Deployment.class);
+    }
+
+    @Test
+    void deploymentRelatedAnnotationsExposeTheirConfiguredValues() throws Exception {
+        Method annotatedDeployment = DeploymentFixture.class.getDeclaredMethod("annotatedDeployment");
+        Method expectedFailureDeployment = DeploymentFixture.class.getDeclaredMethod("expectedFailureDeployment");
+        Method defaultFailureDeployment = DeploymentFixture.class.getDeclaredMethod("defaultFailureDeployment");
+
+        assertThat(annotatedDeployment.getAnnotationsByType(TargetsContainer.class)[0].value()).isEqualTo("managed-server");
+        assertThat(annotatedDeployment.getAnnotationsByType(OverProtocol.class)[0].value()).isEqualTo("Servlet 3.0");
+        assertThat(expectedFailureDeployment.getAnnotationsByType(ShouldThrowException.class)[0].value())
+                .isEqualTo(IllegalStateException.class);
+        assertThat(defaultFailureDeployment.getAnnotationsByType(ShouldThrowException.class)[0].value())
+                .isEqualTo(Exception.class);
+
+        assertRuntimeMethodTarget(OverProtocol.class);
+        assertRuntimeMethodTarget(ShouldThrowException.class);
+        assertThat(TargetsContainer.class.getAnnotationsByType(Retention.class)[0].value()).isEqualTo(RetentionPolicy.RUNTIME);
+        assertThat(TargetsContainer.class.getAnnotationsByType(Target.class)[0].value())
+                .containsExactlyInAnyOrder(ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER);
+    }
+
+    @Test
+    void operationAnnotationsCanTargetMethodsFieldsAndParameters() throws Exception {
+        Method operation = DeploymentFixture.class.getDeclaredMethod("operate", String.class, String.class);
+        Field targetContainer = DeploymentFixture.class.getDeclaredField("targetContainer");
+        Field relatedDeployment = DeploymentFixture.class.getDeclaredField("relatedDeployment");
+        Parameter containerParameter = operation.getParameters()[0];
+        Parameter deploymentParameter = operation.getParameters()[1];
+
+        assertThat(operation.getAnnotationsByType(OperateOnDeployment.class)[0].value()).isEqualTo("orders");
+        assertThat(operation.getAnnotationsByType(RunAsClient.class)[0]).isNotNull();
+        assertThat(targetContainer.getAnnotationsByType(TargetsContainer.class)[0].value()).isEqualTo("managed-server");
+        assertThat(relatedDeployment.getAnnotationsByType(OperateOnDeployment.class)[0].value()).isEqualTo("billing");
+        assertThat(containerParameter.getAnnotationsByType(TargetsContainer.class)[0].value()).isEqualTo("parameter-server");
+        assertThat(deploymentParameter.getAnnotationsByType(OperateOnDeployment.class)[0].value()).isEqualTo("inventory");
+
+        assertThat(OperateOnDeployment.class.getAnnotationsByType(Retention.class)[0].value()).isEqualTo(RetentionPolicy.RUNTIME);
+        assertThat(OperateOnDeployment.class.getAnnotationsByType(Target.class)[0].value())
+                .containsExactlyInAnyOrder(ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER);
+    }
+
+    @Test
+    void runAsClientIsRuntimeVisibleOnMethodsTypesAndInheritedBySubclasses() throws Exception {
+        Method operation = DeploymentFixture.class.getDeclaredMethod("operate", String.class, String.class);
+
+        assertThat(ClientFixture.class.getAnnotationsByType(RunAsClient.class)[0]).isNotNull();
+        assertThat(ClientSubclass.class.getAnnotationsByType(RunAsClient.class)[0]).isNotNull();
+        assertThat(operation.getAnnotationsByType(RunAsClient.class)[0]).isNotNull();
+        assertThat(RunAsClient.class.getAnnotationsByType(Retention.class)[0].value()).isEqualTo(RetentionPolicy.RUNTIME);
+        assertThat(RunAsClient.class.getAnnotationsByType(Target.class)[0].value())
+                .containsExactlyInAnyOrder(ElementType.TYPE, ElementType.METHOD);
+    }
+
+    private static void assertRuntimeMethodTarget(Class<?> annotationType) {
+        assertThat(annotationType.getAnnotationsByType(Retention.class)[0].value()).isEqualTo(RetentionPolicy.RUNTIME);
+        assertThat(annotationType.getAnnotationsByType(Target.class)[0].value()).containsExactly(ElementType.METHOD);
+    }
+
+    private static final class RecordingContainerController implements ContainerController {
+        private final Map<String, Boolean> started = new HashMap<String, Boolean>();
+        private final Map<String, Map<String, String>> configurations = new HashMap<String, Map<String, String>>();
+        private final List<String> events = new ArrayList<String>();
+
+        @Override
+        public void start(String containerQualifier) {
+            start(containerQualifier, Collections.<String, String>emptyMap());
+        }
+
+        @Override
+        public void start(String containerQualifier, Map<String, String> config) {
+            Map<String, String> copiedConfig = new HashMap<String, String>(config);
+            started.put(containerQualifier, Boolean.TRUE);
+            configurations.put(containerQualifier, copiedConfig);
+            events.add("start:" + containerQualifier);
+        }
+
+        @Override
+        public void stop(String containerQualifier) {
+            started.put(containerQualifier, Boolean.FALSE);
+            events.add("stop:" + containerQualifier);
+        }
+
+        @Override
+        public void kill(String containerQualifier) {
+            started.put(containerQualifier, Boolean.FALSE);
+            events.add("kill:" + containerQualifier);
+        }
+
+        @Override
+        public boolean isStarted(String containerQualifier) {
+            return Boolean.TRUE.equals(started.get(containerQualifier));
+        }
+    }
+
+    private static final class RecordingDeployer implements Deployer {
+        private final Set<String> deployed = new HashSet<String>();
+        private final Map<String, byte[]> archives = new HashMap<String, byte[]>();
+        private final List<String> events = new ArrayList<String>();
+
+        void register(String name, String content) {
+            archives.put(name, content.getBytes(StandardCharsets.UTF_8));
+        }
+
+        @Override
+        public void deploy(String name) {
+            deployed.add(name);
+            events.add("deploy:" + name);
+        }
+
+        @Override
+        public InputStream getDeployment(String name) {
+            events.add("get:" + name);
+            return new ByteArrayInputStream(archives.get(name));
+        }
+
+        @Override
+        public void undeploy(String name) {
+            deployed.remove(name);
+            events.add("undeploy:" + name);
+        }
+    }
+
+    private static final class RecordingArchive implements Archive<RecordingArchive> {
+        private final String name;
+        private final Set<String> paths = new HashSet<String>();
+        private final List<String> addedPaths = new ArrayList<String>();
+        private final List<Asset> addedAssets = new ArrayList<Asset>();
+
+        RecordingArchive(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public String getId() {
+            return name;
+        }
+
+        @Override
+        public RecordingArchive add(Asset asset, ArchivePath target) throws IllegalArgumentException {
+            paths.add(target.get());
+            addedPaths.add(target.get());
+            addedAssets.add(asset);
+            return this;
+        }
+
+        @Override
+        public RecordingArchive add(Asset asset, ArchivePath target, String name) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive add(Asset asset, String target, String name) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive add(NamedAsset asset) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive add(Asset asset, String target) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive addAsDirectory(String path) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive addAsDirectories(String... paths) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive addAsDirectory(ArchivePath path) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive addAsDirectories(ArchivePath... paths) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive addHandlers(ArchiveEventHandler... handlers) {
+            return unsupported();
+        }
+
+        @Override
+        public Node get(ArchivePath path) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public Node get(String path) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public <X extends Archive<X>> X getAsType(Class<X> type, String path) {
+            return unsupported();
+        }
+
+        @Override
+        public <X extends Archive<X>> X getAsType(Class<X> type, ArchivePath path) {
+            return unsupported();
+        }
+
+        @Override
+        public <X extends Archive<X>> Collection<X> getAsType(Class<X> type, Filter<ArchivePath> filter) {
+            return unsupported();
+        }
+
+        @Override
+        public <X extends Archive<X>> X getAsType(Class<X> type, String path, ArchiveFormat archiveFormat) {
+            return unsupported();
+        }
+
+        @Override
+        public <X extends Archive<X>> X getAsType(Class<X> type, ArchivePath path, ArchiveFormat archiveFormat) {
+            return unsupported();
+        }
+
+        @Override
+        public <X extends Archive<X>> Collection<X> getAsType(
+                Class<X> type, Filter<ArchivePath> filter, ArchiveFormat archiveFormat) {
+            return unsupported();
+        }
+
+        @Override
+        public boolean contains(ArchivePath path) throws IllegalArgumentException {
+            return paths.contains(path.get());
+        }
+
+        @Override
+        public boolean contains(String path) throws IllegalArgumentException {
+            return paths.contains(path);
+        }
+
+        @Override
+        public Node delete(ArchivePath path) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public Node delete(String path) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public Map<ArchivePath, Node> getContent() {
+            return Collections.emptyMap();
+        }
+
+        @Override
+        public Map<ArchivePath, Node> getContent(Filter<ArchivePath> filter) {
+            return Collections.emptyMap();
+        }
+
+        @Override
+        public RecordingArchive add(Archive<?> archive, ArchivePath path, Class<? extends StreamExporter> exporter)
+                throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive add(Archive<?> archive, String path, Class<? extends StreamExporter> exporter)
+                throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive merge(Archive<?> source) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive merge(Archive<?> source, Filter<ArchivePath> filter) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive merge(Archive<?> source, ArchivePath path) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive merge(Archive<?> source, String path) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive merge(Archive<?> source, ArchivePath path, Filter<ArchivePath> filter)
+                throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive merge(Archive<?> source, String path, Filter<ArchivePath> filter)
+                throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive move(ArchivePath source, ArchivePath target) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public RecordingArchive move(String source, String target) throws IllegalArgumentException {
+            return unsupported();
+        }
+
+        @Override
+        public String toString(boolean verbose) {
+            return name;
+        }
+
+        @Override
+        public String toString(Formatter formatter) throws IllegalArgumentException {
+            return name;
+        }
+
+        @Override
+        public void writeTo(OutputStream outputStream, Formatter formatter) throws IllegalArgumentException {
+            throw new UnsupportedOperationException("Only marker operations are needed by these tests");
+        }
+
+        @Override
+        public Archive<RecordingArchive> shallowCopy() {
+            return unsupported();
+        }
+
+        @Override
+        public <TYPE extends Assignable> TYPE as(Class<TYPE> type) {
+            return unsupported();
+        }
+
+        private static <T> T unsupported() {
+            throw new UnsupportedOperationException("Only marker operations are needed by these tests");
+        }
+    }
+
+    private static final class DeploymentFixture {
+        @TargetsContainer("managed-server")
+        private String targetContainer;
+
+        @OperateOnDeployment("billing")
+        private String relatedDeployment;
+
+        @Deployment
+        static RecordingArchive defaultDeployment() {
+            return new RecordingArchive("default.war");
+        }
+
+        @Deployment(name = "orders", managed = false, order = 7, testable = false)
+        static RecordingArchive manualDeployment() {
+            return new RecordingArchive("orders.war");
+        }
+
+        @Deployment(name = "protocol")
+        @TargetsContainer("managed-server")
+        @OverProtocol("Servlet 3.0")
+        static RecordingArchive annotatedDeployment() {
+            return new RecordingArchive("protocol.war");
+        }
+
+        @Deployment(name = "expectedFailure")
+        @ShouldThrowException(IllegalStateException.class)
+        static RecordingArchive expectedFailureDeployment() {
+            return new RecordingArchive("failure.war");
+        }
+
+        @Deployment(name = "defaultFailure")
+        @ShouldThrowException
+        static RecordingArchive defaultFailureDeployment() {
+            return new RecordingArchive("default-failure.war");
+        }
+
+        @RunAsClient
+        @OperateOnDeployment("orders")
+        void operate(@TargetsContainer("parameter-server") String container,
+                @OperateOnDeployment("inventory") String deployment) {
+        }
+    }
+
+    @RunAsClient
+    private static class ClientFixture {
+    }
+
+    private static final class ClientSubclass extends ClientFixture {
+    }
+}

--- a/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java
+++ b/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/src/test/java/org_jboss_arquillian_container/arquillian_container_test_api/Arquillian_container_test_apiTest.java
@@ -486,6 +486,11 @@ public class Arquillian_container_test_apiTest {
         }
 
         @Override
+        public RecordingArchive filter(Filter<ArchivePath> filter) {
+            return unsupported();
+        }
+
+        @Override
         public RecordingArchive move(ArchivePath source, ArchivePath target) throws IllegalArgumentException {
             return unsupported();
         }
@@ -512,6 +517,11 @@ public class Arquillian_container_test_apiTest {
 
         @Override
         public Archive<RecordingArchive> shallowCopy() {
+            return unsupported();
+        }
+
+        @Override
+        public Archive<RecordingArchive> shallowCopy(Filter<ArchivePath> filter) {
             return unsupported();
         }
 

--- a/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/user-code-filter.json
+++ b/tests/src/org.jboss.arquillian.container/arquillian-container-test-api/1.1.11.Final/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.jboss.arquillian.container.test.api.**"
+    },
+    {
+      "includeClasses" : "org_jboss_arquillian_container.arquillian_container_test_api.**"
+    }
+  ]
+}

